### PR TITLE
feat: oracle should allow configurable collateral types

### PR DESCRIPTION
**Pull request by gregdhill**
Signed-off-by: Gregory Hill <gregorydhill@outlook.com>

### CODE REVIEW

File: oracle/src/main.rs [hunk 1]
```diff
@@ -21,23 +21,77 @@ const NAME: &str = env!("CARGO_PKG_NAME");
 const ABOUT: &str = env!("CARGO_PKG_DESCRIPTION");
 
 const CONFIRMATION_TARGET: u32 = 1;
+
 const BTC_DECIMALS: u32 = 8;
-const DOT_DECIMALS: u32 = 10;
+const BTC_CURRENCY: &str = "btc";
+
+macro_rules! create_collateral_id {
```
Comment (sander2) [inline oracle/src/main.rs:28]: Can't we reuse the primitives crate?

Comment (gregdhill) [reply oracle/src/main.rs:28]: We could but there are two issues:
- `CollateralId`is a subset
- We may have different ids for alternate APIs

Comment (sander2) [reply oracle/src/main.rs:28]: While true that it's a subset, we'll have to use the CurrencyId in the dispatchable functions/rpcs, so we'll have to deal with two very similar types, and convert between them when necessary. Plus we'd be duplicating information, introducing the possibility for mismatches. You really think that's worth it?

**Review by sander2 — APPROVED** [review]

**Review by sander2 — APPROVED** [review]

File: oracle/src/main.rs [hunk 2]
```diff
@@ -21,23 +21,45 @@ const NAME: &str = env!("CARGO_PKG_NAME");
 const ABOUT: &str = env!("CARGO_PKG_DESCRIPTION");
 
 const CONFIRMATION_TARGET: u32 = 1;
+
 const BTC_DECIMALS: u32 = 8;
-const DOT_DECIMALS: u32 = 10;
+const BTC_CURRENCY: &str = "btc";
+
+struct CurrencyIdArg(CurrencyId);
```
Comment (sander2) [inline oracle/src/main.rs:28]: I have a slight preference to use `CurrencyId` as arg directly, and to specify a parsing function. But doesn't have to block merging

**Review by sander2 — APPROVED** [review]

**Pull request merged**